src/chunk.c

```diff
@@ -1691,11 +1692,31 @@ chunk_tuple_found(TupleInfo *ti, void *arg)
 	 * ts_chunk_build_from_tuple_and_stub() since chunk_resurrect() also uses
 	 * that function and, in that case, the chunk object is needed to create
 	 * the data table and related objects. */
-	chunk->table_id = get_relname_relid(chunk->fd.table_name.data,
-										get_namespace_oid(chunk->fd.schema_name.data, true));
+	Oid namespace_oid = get_namespace_oid(chunk->fd.schema_name.data, true);
+
+	Ensure(OidIsValid(namespace_oid),
+		   "namespace for schema %s has invalid OID",
+		   quote_identifier(NameStr(chunk->fd.schema_name)));
+
+	chunk->table_id = get_relname_relid(chunk->fd.table_name.data, namespace_oid);
+	Ensure(OidIsValid(chunk->table_id),
+		   "namespace for chunk %s.%s has invalid OID",
+		   quote_identifier(NameStr(chunk->fd.schema_name)),
+		   quote_identifier(NameStr(chunk->fd.table_name)));
```

Comment:
```suggestion
	chunk->table_id = get_relname_relid(chunk->fd.table_name.data, namespace_oid);
	Ensure(OidIsValid(chunk->table_id),
		   "namespace for chunk \"%s.%s\" has invalid OID",
		   NameStr(chunk->fd.schema_name),
		   NameStr(chunk->fd.table_name));
```

quote_identifier is used for quoting identifier names in a sql context, but since is not a sql context but goes through ereport quote_identifier shouldnt be used